model/queue_discovery.go

```diff
@@ -14,26 +13,18 @@ type TaskQueueAccessor interface {
 // may query/select a task from an existing task queue to support
 // out-of-order task execution for the purpose of task-groups.
 type TaskSpec struct {
-	Group        string
-	BuildVariant string
-	ProjectID    string
-	Version      string
+	Group         string
+	BuildVariant  string
+	ProjectID     string
+	Version       string
+	GroupMaxHosts int
 }
 
-// MatchingOrNextTask provides a generic implementation of logic to
-// support selecting either a task matching the given task spec or the
-// next task in the queue if none matches.
-//
-// Returns nil if the queue is empty.
-func MatchingOrNextTask(queue TaskQueueAccessor, spec TaskSpec) *TaskQueueItem {
+// GetNextTask returns nil if the queue is empty, otherwise delegates to
+// the queue's FindNextTask() method.
+func GetNextTask(queue TaskQueueAccessor, spec TaskSpec) *TaskQueueItem {
```

Comment:
Why don't we just get rid of this entirely and maybe add the short circut for empty queues to FindNextTask.